lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -124,16 +140,24 @@ def exception_class_name(exception=@exception)
         end
 
         def failure_lines
-          @failure_lines ||=
-            begin
-              lines = []
-              lines << failure_slash_error_line unless (description == failure_slash_error_line)
-              lines << "#{exception_class_name}:" unless exception_class_name =~ /RSpec/
-              encoded_string(exception.message.to_s).split("\n").each do |line|
-                lines << "  #{line}"
-              end
-              lines
-            end
+          @failure_lines ||= failure_slash_error_lines + exception_lines
+        end
+
+        def failure_slash_error_lines
+          read_failed_lines.tap do |lines|
+            least_indentation = lines.map { |line| line.match(/^[ \t]*/)[0] }.min
+            lines.map! { |line| '  ' + line.gsub(/^#{least_indentation}/, '') }
```

Comment:
Also, the `map!` (as opposed to `map`) call concerns me a little bit -- since the `lines` array wasn't instantiated locally in this method it's not clear to me from reading the code who else may have a reference to the array and it always makes me nervous to mutate an object that someone else may have a reference to.